service/history/historyEngineInterfaces.go

```diff
@@ -71,4 +72,11 @@ type (
 		common.Daemon
 		NotifyNewTimer(timerTask []persistence.Task)
 	}
+
+	historyEventNotifier interface {
+		common.Daemon
+		NotifyNewHistoryEvent(event *historyEvent)
+		WatchHistoryEvent(domainID *string, execution *workflow.WorkflowExecution) (*string, chan *historyEvent, error)
```

Comment:
Let's not use pointers for primitive types.

Reply:
ok